github_metrics/routes/api/webhooks.py

```
@@ -0,0 +1,189 @@
+    if end_datetime:
+        query += " AND created_at <= $" + str(param_idx)
+        params.append(end_datetime)
+        param_idx += 1
+
+    offset = (page - 1) * page_size
+    if offset > MAX_OFFSET:
+        raise HTTPException(
+            status_code=http_status.HTTP_400_BAD_REQUEST,
+            detail=f"Pagination offset exceeds maximum ({MAX_OFFSET}). Use time filters to narrow results.",
+        )
+    count_query = "SELECT COUNT(*) FROM (" + query + ") AS filtered"
```

Ruff (0.14.7)

Possible SQL injection vector through string-based query construction

(S608)